src/authentication.jl

```diff
@@ -194,14 +202,78 @@ The returned [`Authentication`](@ref) object is also cached globally (overwritin
 cached authentications), making it unnecessary to pass the returned object manually to other
 function calls. This is useful for interactive use, but should not be used in library code,
 as different authentication calls may clash.
+
+# Project Context
+
+An [`Authentication`](@ref) object can also specify the default JuliaHub project.
+This can be set by passing the optional `project` argument, which works as follows:
+
+- If the `project` value is not passed, JuliaHub.jl will attempt to pick up the the project UUID
+  from the `JULIAHUB_PROJECT_UUID` environment variable, and will fall back to the non-project
+  context if that is not set.
+
+- If you pass an explicit UUID (either as a string or an `UUID` object), that will then be used
+  as the project. Note that a UUID passed as a string must be a syntactically correct UUID.
+
+- If you pass `nothing`, that make JuliaHub.jl ignore any values in the `JULIAHUB_PROJECT_UUID`
+  environment variable.
```

Comment:
Thoughts on making the `missing` vs `nothing` distinction more obvious? E.g. a `NO_PROJECT` const in the module or something like that?

Reply:
Went with passing `project = _juliahub_project(missing)`  as the default kwarg value, so that we don't have to accept `Missing`. Passing `UUID` or `Nothing` back into `_juliahub_project` is essentially a no-op, so no need for any other changes.